component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java

```diff
@@ -0,0 +1,305 @@
+                                "Value of this will be in [0,1]. 0 means only old data used and" +
+                                "1 will mean that only new data is used",
+                        optional = true,
+                        type = {DataType.FLOAT},
+                        defaultValue = "0.01f"
+                ),
+                @Parameter(
+                        name = "number.of.clusters",
+                        description = "The assumed number of natural clusters (numberOfClusters) in the data set.",
+                        type = {DataType.INT}
+                ),
+                @Parameter(
+                        name = "coordinate.values",
+                        description = "This is a variable length argument. Depending on the dimensionality of data " +
+                                "points we will receive coordinates along each axis.",
+                        type = {DataType.DOUBLE} //how to give multiple types?
+                )
+
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "euclideanDistanceToClosestCentroid",
+                        description = "Represents the Euclidean distance between the current data point and the " +
+                                "closest centroid.",
+                        type = {DataType.DOUBLE}
+                ),
+                @ReturnAttribute(
+                        name = "closestCentroidCoordinatei",
+                        description = "This is a variable length attribute. Depending on the dimensionality(d) " +
+                                "we will return closestCentroidCoordinate1 to closestCentroidCoordinated",
+                        type = {DataType.DOUBLE}
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "from InputStream#streamingML:KMeansIncremental(modelName, numberOfClusters, " +
+                                "coordinateValue1, coordinateValue2)\"\n" +
+                                "select coordinateValue1, coordinateValue2, euclideanDistanceToClosestCentroid, " +
+                                "closestCentroidCoordinate1, closestCentroidCoordinate2\"\n" +
+                                "insert into OutputStream",
+                        description = "modelName='model1', numberOfClusters=2, This will select first two " +
+                                "distinct values in " +
+                                "events as initial model and predict and update the model for every event"
+                ),
+        }
+)
+public class KMeansIncrementalSPExtension extends StreamProcessor {
+    private float decayRate;
+
+
+    private int numberOfEventsReceived;
+    private int coordinateStartIndex;
+    private LinkedList<DataPoint> dataPointsArray;
+    private double[] coordinateValuesOfCurrentDataPoint;
+
+    private boolean modelTrained = false;
```

Comment:
Default value of a boolean is false. You don't need to initialize to false.

Reply:
done